FrontEnd/src/UserAuthentication/Auth.js

```diff
@@ -6,17 +6,32 @@ import {
   onAuthStateChanged,
   sendPasswordResetEmail,
 } from "firebase/auth";
+import { useState, useEffect } from "react";
+import { addUser, isUsernameUnique } from "./FirestoreDB";
 
-const registerUser = async (email, password) => {
+const registerUser = async (email, password, username, firstName, lastName) => {
   try {
-    const user = await createUserWithEmailAndPassword(auth, email, password);
-    if (user!==null){
-      return user;
-    };
+    const isUnique = await isUsernameUnique(username);
+    if (isUnique !== true) {
+      alert("Username taken");
```

Comment:
Suggested improvement (not for this commit, but can add it to the "good to have" list) - show an error message next to the username input in the signup form